File: src/main/Java/boj_step14/boj14425.java
import java.io.*;
import java.util.HashSet;
import java.util.StringTokenizer;

public class boj14425 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int m = Integer.parseInt(tokenizer.nextToken());
        HashSet<String> keySet = new HashSet<>();
        int count = 0;
        while (n-- > 0) keySet.add(reader.readLine());
        while (m-- > 0) if (keySet.contains(reader.readLine())) count++;
        writer.append(String.valueOf(count)).close();
    }
}

File: src/main/Kotlin/boj_step14/boj14425.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (n, m) = r.readLine().split(" ").map { it.toInt() }
    val set = mutableSetOf<String>()
    val list = mutableListOf<String>()
    var count = 0
    repeat(n) {
        set.add(r.readLine())
    }
    repeat(m) {
        list.add(r.readLine())
    }
    list.forEach {
        if (set.contains(it)) count++
//        if (!set.add(it)) count++
    }
    w.write(count.toString())
    w.close()
}